### PR TITLE
Cmake option to disable proxy

### PR DESCRIPTION
**Pull request by allexzander**
Signed-off-by: alex-z <blackslayer4@gmail.com>

This will fix https://github.com/nextcloud/desktop/issues/4138

### CODE REVIEW

**Comment by allexzander** [comment]
@tobiasKaminsky This will require adding the corresponding `DO_NOT_USE_PROXY` on the PHP side and setting it to ON when needed.

**Review by claucambra — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
👍
As I have no real clue about C++/QT: is the proxy setting then also hidden in settings menu?

**Comment by allexzander** [comment]
> 👍 As I have no real clue about C++/QT: is the proxy setting then also hidden in settings menu?

@tobiasKaminsky Yes, hidden. Also, the config file is overridden and a "No Poxy" setting is written to it.

File: src/gui/application.cpp [hunk 1]
```diff
@@ -250,6 +250,12 @@ Application::Application(int &argc, char **argv)
         }
     }
 
+    if (_theme->doNotUseProxy()) {
+        ConfigFile().setProxyType(QNetworkProxy::NoProxy);
+    }
+
+    ConfigFile cfgFile;
```
Comment (mgallien) [inline src/gui/application.cpp:257]: why is this line needed ?

Comment (mgallien) [reply src/gui/application.cpp:257]: you seem not to use `configFile` variable

Comment (allexzander) [reply src/gui/application.cpp]: @mgallien thx, and, fixed

**Review by mgallien — APPROVED** [review]
